src/app/components/core/south/south-service-modal/south-service-modal.component.ts

```diff
@@ -263,6 +271,42 @@ export class SouthServiceModalComponent implements OnInit, OnChanges {
     this.child.toggleModal(true);
   }
 
+  getAssetReadings(service) {
+    console.log('download start time', performance.now());
+    const fields = ['assetName', 'reading', 'timestamp'];
+    const opts = { fields };
+    const assets = service.assets;
+    let assetReadings = [];
+    assets.forEach((ast, i) => {
+      this.assetService.getAssetReadings(encodeURIComponent(ast.asset), ast.count).
+        subscribe(
+          (result: any[]) => {
+            result = result.map(r => {
+              r['assetName'] = ast.asset;
+              return r;
+            });
+            assetReadings = assetReadings.concat(result);
+            if (i === assets.length - 1) {
+              const parser = new Parser(opts);
+              const csv = parser.parse(assetReadings);
+              const blob = new Blob([csv], { type: 'text/csv' });
+              const url = window.URL.createObjectURL(blob);
+              // create a custom anchor tag
+              const a = document.createElement('a');
+              a.href = url;
+              a.download = service['name'] + '-readings.csv';
```

Comment:
Please keep it consistent as with assetCode. Either use `-` or `_` at both the places!